### DISCUSSION TITLE
On fitting the model with the data, I am getting the below error and unable to figure out the reason. Need Help.

### DISCUSSION THREAD

**Discussion by priyankabudavi** [MaartenGr/BERTopic#1605] [General]
My data : I tried with dataframe it didnt work,

combined_text = ['high risk client matter ongoing monitoring form law firm client due diligence ',
 'counter notice  premises in wales  ',
 'witness statement supporting application to remove arbitrator  failure properly to conduct proceedings ',
 'order cancelling or confirming variation of rights of holders of special class of shares ',
 'transfer whole unregistered freehold land subject to a charge seller not released indemnity from buyer option for direct covenant by buyer ',
 'clause gift of a remainder interest other than to the life tenant deed of assignment ',
 'shared parental leave booking form  period of leave notice   birth  form for mothers  or prospective mothers   fathers  or prospective fathers   husbands  wives  civil partners or partners seeking shared parental leave  used to make a request to the employer for a firm booking of one or more periods of leave' ]

I believe it fails while clustering the data when  HDBSCAN  is invoked. 

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File [~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\bertopic\_bertopic.py:3218](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/budavip/DBSCAN/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0/LocalCache/local-packages/Python311/site-packages/bertopic/_bertopic.py:3218), in BERTopic._cluster_embeddings(self, umap_embeddings, documents, partial_fit, y)
   3217 try:
-> 3218     self.hdbscan_model.fit(umap_embeddings, y=y)
   3219 except TypeError:

File [~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\hdbscan\hdbscan_.py:1205](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/budavip/DBSCAN/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0/LocalCache/local-packages/Python311/site-packages/hdbscan/hdbscan_.py:1205), in HDBSCAN.fit(self, X, y)
   1196 kwargs.update(self._metric_kwargs)
   1198 (
   1199     self.labels_,
   1200     self.probabilities_,
   1201     self.cluster_persistence_,
   1202     self._condensed_tree,
   1203     self._single_linkage_tree,
   1204     self._min_spanning_tree,
-> 1205 ) = hdbscan(clean_data, **kwargs)
   1207 if self.metric != "precomputed" and not self._all_finite:
   1208     # remap indices to align with original data in the case of non-finite entries.

File [~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\hdbscan\hdbscan_.py:884](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/budavip/DBSCAN/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0/LocalCache/local-packages/Python311/site-packages/hdbscan/hdbscan_.py:884), in hdbscan(X, min_cluster_size, min_samples, alpha, cluster_selection_epsilon, max_cluster_size, metric, p, leaf_size, algorithm, memory, approx_min_span_tree, gen_min_span_tree, core_dist_n_jobs, cluster_selection_method, allow_single_cluster, match_reference_implementation, **kwargs)
    868             (single_linkage_tree, result_min_span_tree) = memory.cache(
    869                 _hdbscan_boruvka_balltree
    870             )(
   (...)
...
File hdbscan\\_hdbscan_tree.pyx:43, in hdbscan._hdbscan_tree.condense_tree()

File hdbscan\\_hdbscan_tree.pyx:109, in hdbscan._hdbscan_tree.condense_tree()

TypeError: 'numpy.float64' object cannot be interpreted as an integer

**Comment by MaartenGr** [comment]
It's difficult to say without seeing your code but I guess this is a result of simply having too few data points. The default `min_topic_size` is 10 I believe which means that clustering < 10 documents is not possible.
